### DISCUSSION TITLE
Can I add a slider/horizontal bar over a colorbar?

### DISCUSSION THREAD

**Discussion by yagomg21** [JuliaPlots/Makie.jl#3614] [Q&A]
Hello! I'm a noob with Makie and I am trying to use it to show a 3d trajectory, including rotations. I am quite happy with what it is looking now, I am using a mesh object inside Axis3 and rotating it so that it represents the rotations in the animations. I have a few questions regarding this.

The first one is, when I use autolimits!(), is there any way of telling it to maintain a certain aspect ratio?

Secondly, I am thinking of making an Observable for a vector's magnitude, and showing the changes through a heatmap, changing its color, and including a colorbar next to it for reference. As it is an animation, I thought it would be very nice if I could have some kind of slider showing the value on the colorbar as well. Is there a way of doing this?

Finally, is it possible to use observables with rotate!(), to make it rotate whenever the observable changes?

Thank you very much for your help!

**Comment by asinghvi17** [comment]
1. Yes - you can set the `aspect` attribute a la `ax3.aspect`, or as a keyword argument during construction.

2. Maybe, kind of.   You could potentially create a second colorbar on top of the first, and set its colormap such that it has e.g. 100 colors and only one is visible at a time?  You could take inspiration from e.g. https://www.linkedin.com/posts/alex-s-gardner_greenland-julia-itsabrlive-activity-7056483156798947328-99pH?utm_source=share&utm_medium=member_desktop

3. Yes - you can `lift` on the observable and call `rotate!` inside that lift, so it will set the rotation whenever the observable triggers.
